px4_ros2_cpp/test/integration/local_navigation.cpp

```diff
@@ -0,0 +1,163 @@
+ */
+
+#include <gtest/gtest.h>
+#include <rclcpp/rclcpp.hpp>
+#include <px4_msgs/msg/estimator_status_flags.hpp>
+#include <px4_ros2/navigation/experimental/local_position_measurement_interface.hpp>
+#include "util.hpp"
+
+using namespace std::chrono_literals;
+using px4_ros2::LocalPositionMeasurement, px4_ros2::LocalPositionMeasurementInterface;
+using px4_msgs::msg::EstimatorStatusFlags;
+
+class LocalPositionInterfaceTest : public BaseTest
+{
+protected:
+  void SetUp() override
+  {
+    _node = initNode();
+
+    _local_navigation_interface = std::make_shared<LocalPositionMeasurementInterface>(
+      *_node, px4_ros2::PoseFrame::LocalNED,
+      px4_ros2::VelocityFrame::LocalNED);
+    ASSERT_TRUE(_local_navigation_interface->doRegister()) <<
+      "Failed to register LocalPositionMeasurementInterface.";
+
+    // Subscribe to PX4 EKF estimator status flags
+    _subscriber = _node->create_subscription<EstimatorStatusFlags>(
+      "/fmu/out/estimator_status_flags", rclcpp::QoS(10).best_effort(),
+      [this](EstimatorStatusFlags::UniquePtr msg) {
+        _estimator_status_flags = std::move(msg);
+      });
+  }
+
+  void resetFlags()
+  {
+    // Wait for PX4 to stop fusing external measurements
+    while (_estimator_status_flags->cs_ev_pos || _estimator_status_flags->cs_ev_hgt ||
+      _estimator_status_flags->cs_ev_vel || _estimator_status_flags->cs_ev_yaw)
+    {
+      rclcpp::spin_some(_node);
+      std::this_thread::sleep_for(_k_sleep_interval);
+    }
+
+    _estimator_status_flags.reset();
+  }
+
+  void waitForMeasurementUpdate(
+    std::unique_ptr<LocalPositionMeasurement> measurement,
+    const std::function<bool(const EstimatorStatusFlags::UniquePtr &)> & is_fused_getter
+  )
+  {
+    auto start_time = std::chrono::steady_clock::now();
+
+    // Wait for PX4 estimator flags to confirm proper measurement fusion
+    while (!_estimator_status_flags || !is_fused_getter(_estimator_status_flags)) {
+
+      // Send measurement
+      measurement->timestamp_sample = _node->get_clock()->now();
+      ASSERT_NO_THROW(
+        _local_navigation_interface->update(*measurement)
+      ) << "Failed to send position measurement update via LocalPositionMeasurementInterface.";
+
+      rclcpp::spin_some(_node);
+
+      // Check timeout
+      const auto current_time = std::chrono::steady_clock::now();
+      const auto elapsed_time = std::chrono::duration_cast<std::chrono::milliseconds>(
+        current_time - start_time
+      );
+
+      if (elapsed_time >= _k_timeout_duration) {
+        ASSERT_NE(_estimator_status_flags, nullptr) <<
+          "Missing feedback from PX4: no estimator status flags published over /fmu/out/estimator_status_flags.";
+        EXPECT_TRUE(is_fused_getter(_estimator_status_flags)) <<
+          "Position measurement update was not fused into the PX4 EKF.";
+        break;
+      }
+
+      std::this_thread::sleep_for(_k_sleep_interval);
+    }
+
+    resetFlags();
```

Comment:
How long does that take? If it takes a while, you could also do it only in one of the tests.
You could also rename this to reflect a bit better what it does (something like `waitUntilFlagsReset()`)

Reply:
Less than one second normally. But we would still need to wait between each test for the estimator flags to reset, otherwise they might affect the result of the next test